tests/test_auth.py

```diff
@@ -7,29 +7,51 @@
 from whirlpool.auth import Auth
 
 from .aiohttp import AiohttpClientMocker
-from .mock_backendselector import BackendSelectorMock
+from .mock_backendselector import BackendSelectorMock, BackendSelectorMockMultipleCreds
 
-BACKEND_SELECTOR_MOCK = BackendSelectorMock()
-
-AUTH_URL = f"{BACKEND_SELECTOR_MOCK.base_url}/oauth/token"
-AUTH_DATA = {
-    "client_id": BACKEND_SELECTOR_MOCK.client_id,
-    "client_secret": BACKEND_SELECTOR_MOCK.client_secret,
-    "grant_type": "password",
-    "username": "email",
-    "password": "secretpass",
-}
 AUTH_HEADERS = {
     "Content-Type": "application/x-www-form-urlencoded",
 }
 
+BACKEND_SELECTORS = [BackendSelectorMockMultipleCreds(), BackendSelectorMock()]
 
 pytestmark = pytest.mark.asyncio
 
 
-async def test_auth_success(http_client_mock: AiohttpClientMocker):
+@pytest.mark.parametrize("refresh_token", [None, "refresh_token_123"])
+@pytest.mark.parametrize("backend_selector", BACKEND_SELECTORS)
+async def test_get_auth_body(
```

Comment:
I think we can keep the check for the auth_data on `test_auth_success` as before, and remove this test, so that we use only public methods here too.
We simply need to construct the auth_data dict from the first element on `backend_selector.client_credentials`.
We could also add another test `test_auth_multiple_client_credentials` where the http_client_mock returns 404 for each call, and we just check that it goes trough all the `backend_selector.client_credentials`.